core/src/main/java/hudson/util/AbstractTaskListener.java

```diff
@@ -1,11 +1,16 @@
 package hudson.util;
 
+import hudson.RestrictedSince;
 import hudson.model.TaskListener;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 
 /**
  * @deprecated implement {@link TaskListener} directly
  */
 @Deprecated
+@Restricted(NoExternalUse.class)
```

Comment:
Overkill. Was already deprecated.

Reply:
People tend to ignore deprecation warnings though.

Reply:
If we want to kill it, I see no problem with overkilling

Reply:
problem is that killing it means you need to use the TaskListener.
TaskListener did not have the morthods in prior versions.

Pipelines checking compatability with current and newest cores will break here and there is no easy solution.  This makes me a sad panda.

Deprecated code should not be restricted - it should be deprecated, then later restricted then later removed IMO.  Needs a JEP.

Reply:
Maybe. I'd guess it makes sense to create a revert PR

Reply:
> Pipelines checking compatability with current and newest cores will break here

If there were any such. `plugin-compat-tester` ignores source compatibility.

Anyway

> Deprecated code should not be restricted - it should be deprecated

Yes. If people ignore deprecation warnings, and we have serious deprecations we really need people to stop using by a particular time, we should be addressing that in `buildPlugin.groovy`, or `plugin-pom`, or grepping for usages, or searching for PCT failures, or something. But

> then later restricted then later removed

Or simply removed. There is no need for `@Restricted` in this context at all. There is only one purpose for `@Restricted`: _introducing_ a member which for technical reasons must be `public` (for example, due to reflective calls) but which you do not actually want or expect to be referenced statically. It is a workaround for the lack of a module system in Java 8-. It has nothing to do with deprecation. IMO.

Reply:
@jglick 
>> Pipelines checking compatibility with current and newest cores will break here

> If there were any such. `plugin-compat-tester` ignores source compatibility.

They exist, they are just not open source.  PCT is checking binary compatibility.  This closed source one checks source compatibility as there is a need to check that as well.

Whilst sometimes source compatibility is broken for a reasonable reason - in this case it seems a bit pointless, as there is no way to be source compatible with the earlier LTS and later weekly versions.

>> then later restricted then later removed

> Or simply removed. 

Something we can discuss in the future JEP I will be writing :-)

Reply:
FTR this just reared its ugly head _again_. `maven-plugin` cannot be compiled against weeklies because of this. And we cannot change its code to just implement `TaskListener`, because then it will not work with its original baseline. So we need to copy the `hyperlink` implementation. Irritating problem of our own making.

Reply:
@jglick could you propose a PR restoring it?